src/main/java/com/aws/greengrass/shadowmanager/sync/IotDataPlaneClientFactory.java

```diff
@@ -125,4 +111,38 @@ private void configureClient(DeviceConfiguration deviceConfiguration) {
         }
         this.iotDataPlaneClient = iotDataPlaneClientBuilder.build();
     }
+
+    /**
+     * Getter for IoT data plane client. This configures the client everytime the getter is used.
+     *
+     * @return iotDataPlaneClient
+     * @throws IoTDataPlaneClientCreationException exception during client configuration
+     */
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException"})
+    public IotDataPlaneClient getIotDataPlaneClient() throws IoTDataPlaneClientCreationException {
+        try {
+            configureClient();
+        } catch (Exception e) {
+            throw new IoTDataPlaneClientCreationException(e);
+        }
+        return this.iotDataPlaneClient;
+    }
+
+
+    /**
+     * Checks if the crypto key service provider is available by getting key managers.
+     *
+     * @throws Exception exception during retry
+     */
+    @SuppressWarnings({"PMD.SignatureDeclareThrowsException"})
+    public void waitForCryptoKeyServiceProvider() throws Exception {
+        logger.atDebug().log("Checking if the crypto key service provider is available");
+        RetryUtils.RetryConfig retryConfig =
+                RetryUtils.RetryConfig.builder().initialRetryInterval(Duration.ofSeconds(2)).maxAttempt(3)
```

Comment:
do we really want to retry here? Or should we let it retry at the higher level?

I'm alright with this, just asking.

Reply:
If we want to retry only for `TLSAuthException` type, this would be cleaner.

Reply:
You're throwing TLSAuthException up to the caller too, so they can also retry on that specific error if they want.